mongodbatlas/resource_mongodbatlas_cluster_migrate.go

```diff
@@ -0,0 +1,340 @@
+							Optional: true,
+							Default:  "ZoneName managed by Terraform",
+						},
+					},
+				},
+			},
+			"mongo_db_version": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"mongo_uri": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"mongo_uri_updated": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"mongo_uri_with_options": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"paused": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"srv_address": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"state_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"advanced_configuration": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"fail_index_key_too_long": {
+							Type:     schema.TypeBool,
+							Optional: true,
+							Computed: true,
+						},
+						"javascript_enabled": {
+							Type:     schema.TypeBool,
+							Optional: true,
+							Computed: true,
+						},
+						"minimum_enabled_tls_protocol": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Computed: true,
+						},
+						"no_table_scan": {
+							Type:     schema.TypeBool,
+							Optional: true,
+							Computed: true,
+						},
+						"oplog_size_mb": {
+							Type:     schema.TypeInt,
+							Optional: true,
+							Computed: true,
+						},
+						"sample_size_bi_connector": {
+							Type:     schema.TypeInt,
+							Optional: true,
+							Computed: true,
+						},
+						"sample_refresh_interval_bi_connector": {
+							Type:     schema.TypeInt,
+							Optional: true,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"labels": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Set: func(v interface{}) int {
+					var buf bytes.Buffer
+					m := v.(map[string]interface{})
+					buf.WriteString(m["key"].(string))
+					buf.WriteString(m["value"].(string))
+					return hashcode.String(buf.String())
+				},
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"key": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Computed: true,
+						},
+						"value": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"snapshot_backup_policy": computedCloudProviderSnapshotBackupPolicySchema(),
+		},
+	}
+}
+
+func resourceMongoDBAtlasClusterStateUpgradeV0(rawState map[string]interface{}, meta interface{}) (map[string]interface{}, error) {
+	log.Println("[INFO] Found MongoDB Cluser state v0; migrating to v1")
+	rawState = migrateAdvancedConfiguration(rawState)
+
+	return rawState, nil
+}
+
+func migrateAdvancedConfiguration(rawState map[string]interface{}) map[string]interface{} {
+	rawState["advanced_configuration"] = []interface{}{}
```

Comment:
I have a question, could you test it using `TF_CLI_ARGS_plan="-refresh=false"` to see if we got any error?

Reply:
Otherwise LGTM! Nice catch!

Reply:
@marin this is what happens:
**state v0.5.1**
```
"       advanced_configuration": {
              "fail_index_key_too_long": "true",
              "javascript_enabled": "true",
              "minimum_enabled_tls_protocol": "TLS1_2",
              "no_table_scan": "false",
              "oplog_size_mb": "",
              "sample_refresh_interval_bi_connector": "",
              "sample_size_bi_connector": ""
            },
```
**using this provider version**
terraform plan -out plan -refresh=false
terraform apply plan -refresh=false

```
            "advanced_configuration": [
              {
                "fail_index_key_too_long": true,
                "javascript_enabled": true,
                "minimum_enabled_tls_protocol": "TLS1_2",
                "no_table_scan": false,
                "oplog_size_mb": 0,
                "sample_refresh_interval_bi_connector": 0,
                "sample_size_bi_connector": 0
              }
            ],
```